client/app/dbaas/logs/detail/options/logs-options.service.js

```diff
@@ -61,26 +61,58 @@ class LogsOptionsService {
     }
 
     /**
-     * returns the list of options that have been subscribed in the service
+     * makes API call to get the list of options that have been subscribed in the service
      *
      * @param {any} serviceName
      * @returns promise that resolves with the array of options which have been subscribed
      * @memberof LogsOptionsService
      */
     getSubscribedOptions (serviceName) {
-        const transformSubscribedOption = this.transformSubscribedOption.bind(this);
         return this.OvhApiDbaasLogs.Accounting().Aapi().me({
             serviceName
-        }).$promise
+        })
+            .$promise
+            .catch(this.ServiceHelper.errorHandler("logs_options_current_options_loading_error"));
+    }
+
+    /**
+     * returns map of all subscribed options wiht there count.
+     *
+     * @param {any} serviceName
+     * @returns map of subscribed option with count
+     * @memberof LogsOptionsService
+     */
+    getSubscribedOptionsMap (serviceName) {
+        const transformSubscribedOption = this.transformSubscribedOption.bind(this);
+        return this.getSubscribedOptions(serviceName)
             .then(response => {
                 const optionsCountMap = _.reduce(response.options, (optionsMap, option) => {
                     optionsMap[option.reference] = optionsMap[option.reference] ? ++optionsMap[option.reference] : 1;
                     return optionsMap;
                 }, {});
-
                 return _.map(_.keys(optionsCountMap), option => transformSubscribedOption(option, optionsCountMap));
-            })
-            .catch(this.ServiceHelper.errorHandler("logs_options_current_options_loading_error"));
+            });
+    }
+    /**
+     * returns all subscribed options with reference "logs-stream".
+     *
+     * @param {any} serviceName
+     * @returns array of all subscribed option objects belonging to streams
+     * @memberof LogsOptionsService
+     */
+    getStreamSubscribedOptions (serviceName) {
+        return this.getSubscribedOptions(serviceName)
+            .then(response => response.options
+                .filter(option => _.startsWith(option.reference, "logs-stream"))
+                .map(option => {
+                    const type = this.$translate.instant(`${option.reference}-type`);
```

Comment:
I tend to keep variable declaration at a minimum in loops but you do what you want ^^

Reply:
removed variables